docs/book/src/basics/consts.md

```diff
@@ -0,0 +1,92 @@
+# Constants
+
+Constants are similar to variables; however, there are a few differences:
+
+- Constants are always evaluated at compile-time
+- Constants can be declared both inside of a [function](../index.md) and at global / `impl` scope.
+- The `mut` keyword cannot be used with constants.
+
+## Associated Constants
+
+Associated constants are constants associated with a type.
+
+```sway
+const ID: u32 = 0;
+```
+
```

Comment:
This is still a little be general, isn't it?  Associated consts are only declared and/or implemented in (i.e., associated with) traits and impls, yeah?  The above is just a regular non-associated const.  I think that code snippet should go in the previous `#Constants` section.